raiden/transfer/log.py

```diff
@@ -192,6 +224,13 @@ def get_all_state_changes(self):
             for res in self.storage.get_all_state_changes()
         ]
 
+    def get_all_state_events(self):
+        """ Returns a list of tuples of event id, state_change_id and events"""
+        return [
+            (res[0], res[1], self.serializer.deserialize(res[2]))
+            for res in self.storage.get_all_state_events()
+        ]
+
     def snapshot(self, state):
         serialized_data = self.serializer.serialize(state)
-        self.storage.write_state_snapshot(self.last_identifier(), serialized_data)
+        self.storage.write_state_snapshot(self.last_state_change_id(), serialized_data)
```

Comment:
This is also assuming that a single state_change is applied before the snapshot is taken, this must have the `state_change_id` as an argument.